cmd/kubeadm/app/master/manifests.go

```diff
@@ -311,28 +319,11 @@ func getAPIServerCommand(cfg *kubeadmapi.MasterConfiguration, selfHosted bool, k
 		command = []string{"/usr/bin/flock", "--exclusive", "--timeout=30", "/var/lock/api-server.lock"}
 	}
 
-	defaultArguments := map[string]string{
-		"insecure-port":                     "0",
-		"admission-control":                 kubeadmconstants.DefaultAdmissionControl,
-		"service-cluster-ip-range":          cfg.Networking.ServiceSubnet,
-		"service-account-key-file":          path.Join(cfg.CertificatesDir, kubeadmconstants.ServiceAccountPublicKeyName),
-		"client-ca-file":                    path.Join(cfg.CertificatesDir, kubeadmconstants.CACertName),
-		"tls-cert-file":                     path.Join(cfg.CertificatesDir, kubeadmconstants.APIServerCertName),
-		"tls-private-key-file":              path.Join(cfg.CertificatesDir, kubeadmconstants.APIServerKeyName),
-		"kubelet-client-certificate":        path.Join(cfg.CertificatesDir, kubeadmconstants.APIServerKubeletClientCertName),
-		"kubelet-client-key":                path.Join(cfg.CertificatesDir, kubeadmconstants.APIServerKubeletClientKeyName),
-		"secure-port":                       fmt.Sprintf("%d", cfg.API.BindPort),
-		"allow-privileged":                  "true",
-		"experimental-bootstrap-token-auth": "true",
-		"storage-backend":                   "etcd3",
-		"kubelet-preferred-address-types":   "InternalIP,ExternalIP,Hostname",
-		// add options to configure the front proxy.  Without the generated client cert, this will never be useable
-		// so add it unconditionally with recommended values
-		"requestheader-username-headers":     "X-Remote-User",
-		"requestheader-group-headers":        "X-Remote-Group",
-		"requestheader-extra-headers-prefix": "X-Remote-Extra-",
-		"requestheader-client-ca-file":       path.Join(cfg.CertificatesDir, kubeadmconstants.FrontProxyCACertName),
-		"requestheader-allowed-names":        "front-proxy-client",
+	var defaultArguments map[string]string
+	if selfHosted {
```

Comment:
Using a Secret for each TLS pair, we need to modify the arguments to take into account that not all secrets can live in `/etc/kubernetes/pki` since multiple secrets would want to mount at that path.